package com.lightningkite.kommon.exception

/**
 * A list of possible exceptions, used for situations where the exception itself cannot be passed.
 */
object ExceptionNames {
    const val Exception = "Exception"
    const val ForbiddenException = "ForbiddenException"
    const val RuntimeException = "RuntimeException"
    const val IllegalArgumentException = "IllegalArgumentException"
    const val IllegalStateException = "IllegalStateException"
    const val IndexOutOfBoundsException = "IndexOutOfBoundsException"
    const val ConcurrentModificationException = "ConcurrentModificationException"
    const val UnsupportedOperationException = "UnsupportedOperationException"
    const val NumberFormatException = "NumberFormatException"
    const val NullPointerException = "NullPointerException"
    const val ClassCastException = "ClassCastException"
    const val AssertionError = "AssertionError"
    const val NoSuchElementException = "NoSuchElementException"
    const val ElementAlreadyExistsException = "ElementAlreadyExistsException"
    const val ArithmeticException = "ArithmeticException"
}